### ISSUE TITLE
kahan-float returning -nan

### ISSUE THREAD

**Issue by vasconcellospedro** [optframe/kahan-float#1]
The kfloat is returning -nan, everytime INFINITY is assigned to it. (only test with kfloat64)

**Comment by igormcoelho** [comment]
Very interesting bug... just observed it, but still don't know best way to fix it.

Simple example:
float f = INFINITY;
f+=1;
f-=1;
We expect inf, but current code breaks.

```
ADDING '1' TO 'inf'
this->c: '0'
y: '1'
t: '1'
this->c: '-nan'
FINAL VAL: 'inf'

ADDING '-1' TO 'inf'
this->c: '-nan'
y: '-nan'
t: '-nan'
this->c: '-nan'
FINAL VAL: '-nan'
```

**Comment by igormcoelho** [comment]
This was fixed @vasconcellospedro , please see v0.2 :)
